src/components/InputForm/InputForm.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useRef } from 'react';
+import { Errors } from '../../utils/Errors';
+import { handleError } from '../../utils/handleError';
+import { Todo } from '../../types/Todo';
+import { postTodo } from '../../api/todos';
+import { TEMPLATE_TODO } from '../../utils/CONSTANTS';
+
+interface Props {
+  setErrorMessage: (error: Errors) => void;
+  setTempTodo: (todo: Todo | null) => void;
+  onAddTodo: (newTodo: Todo) => void;
+  tempTodo: Todo | null;
+  todosLength: number;
+}
+
+export const InputForm: React.FC<Props> = ({
+  setErrorMessage,
+  onAddTodo,
+  tempTodo,
+  setTempTodo,
+  todosLength,
+}) => {
+  const inputField = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    inputField.current?.focus();
+  }, [tempTodo, todosLength]);
```

Comment:
Move the `useEffect `to the end of code

Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.